src/System.Net.Http/src/System/Net/Http/HttpMethod.cs

```diff
@@ -1,27 +1,24 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using System;
-using System.Collections.Generic;
-using System.Text;
-using System.Threading.Tasks;
-using System.Diagnostics.Contracts;
+using System.Diagnostics;
 
 namespace System.Net.Http
 {
     public class HttpMethod : IEquatable<HttpMethod>
     {
-        private string _method;
-
-        private static readonly HttpMethod s_getMethod = new HttpMethod("GET");
-        private static readonly HttpMethod s_putMethod = new HttpMethod("PUT");
-        private static readonly HttpMethod s_postMethod = new HttpMethod("POST");
-        private static readonly HttpMethod s_deleteMethod = new HttpMethod("DELETE");
-        private static readonly HttpMethod s_headMethod = new HttpMethod("HEAD");
-        private static readonly HttpMethod s_optionsMethod = new HttpMethod("OPTIONS");
-        private static readonly HttpMethod s_traceMethod = new HttpMethod("TRACE");
-
-        // Don't expose CONNECT as static property, since it's used by the transport to connect to a proxy. 
+        private readonly string _method;
+        private int _hashcode;
```

Comment:
I have the same concern about the extra field (I'm still in favor of changing `GetHashCode()` from `_method.ToUpperInvariant().GetHashCode()` to `StringComparer.OrdinalIgnoreCase.GetHashCode(_method)`), and even considered only applying the optimization to the static cached instances through the use of a private subclass, to avoid the extra field.

However, I assume the static cached instances are used the vast majority of the time (e.g. `HttpMethod.Get`, `HttpMethod.Post`, etc. vs. `new HttpMethod("GET")`, `new HttpMethod("POST")`, etc.), either directly or through the `GetAsync` and `PostAsync` convenience methods (which use the cached instances), and more rare custom methods would be created once and cached by any developer who cares about memory use, meaning there would only ever be a small number of `HttpMethod` instances in use, under typical usage. With only a handful of instances, the extra field doesn't matter that much.

My only pause is that some developers may be unaware of the static properties, and when they see a method that takes an `HttpMethod` parameter, they may immediately try to "new one up" and just use the constructor without knowing about the static properties. (It's too late now, but perhaps a better design would have had a static factory method instead of the public constructor, which would guide developers away from "new HttpMethod" and towards discovery of the static properties; not to mention making `HttpMethod` a struct). That said, the `GetAsync` and `PostAsync` convenience methods seem to be pretty commonly used, so developers may never even run into this. There's also a possibility that some developers using a custom method may always create a new instance instead of using a single cached instance. But I suspect these cases are rare.

@davidsh, @CIPop, @SidharthNabar, are these faulty assumptions? Do you have any data on how often the constructor and `GetHashCode` are used?